core/arch/arm/kernel/unwind_arm32.c

```diff
@@ -113,18 +113,28 @@ static int32_t expand_prel31(uint32_t prel31)
  * Perform a binary search of the index table to find the function
  * with the largest address that doesn't exceed addr.
  */
-static struct unwind_idx *find_index(uint32_t addr)
+static struct unwind_idx *find_index(uint32_t addr, uaddr_t exidx,
+				     size_t exidx_sz)
 {
 	vaddr_t idx_start, idx_end;
 	unsigned int min, mid, max;
 	struct unwind_idx *start;
 	struct unwind_idx *item;
 	int32_t prel31_addr;
-	uint32_t func_addr;
+	vaddr_t func_addr;
 
-	start = &__exidx_start;
-	idx_start = (vaddr_t)&__exidx_start;
-	idx_end = (vaddr_t)&__exidx_end;
+	if (exidx && exidx_sz) {
```

Comment:
I'd prefer if `exidx` and `exidx_sz` always where correct, that way we could avoid this special case.